Desktop/analysis/jaspcontrol.cpp

```diff
@@ -191,68 +192,54 @@ void JASPControl::componentComplete()
 	}
 
 	QQmlContext* context = qmlContext(this);
-	bool hasContextForm = context->contextProperty("hasContextForm").toBool();
+	bool isDynamic = context->contextProperty("isDynamic").toBool();
+	_form = context->contextProperty("form").value<AnalysisForm*>();
```

Comment:
Can't the control now simply go up in the tree of parents until it reaches the root or a form?